### PR TITLE
fix(mason_logger): correct J and K key mappings

### LINKED ISSUE #413: fix: `chooseOne` `j` and `k` key navigate in the wrong direction
**Description**

When using `chooseOne` <kbd>J</kbd> and <kbd>K</kbd> to navigate, each move in the wrong direction.

Issue #398 for reference :)

**Steps To Reproduce**

```dart
Future<void> main() async {
  final logger = Logger();

  final favoriteColor = logger.chooseOne(
    'Pick a color.',
    choices: ['red', 'green', 'blue'],
    defaultValue: 'blue',
  );

  logger.info('Got it, $favoriteColor it is!');
}
```

```sh
dart run main.dart
```
Navigate with  <kbd>J</kbd> and <kbd>K</kbd>.

**Expected Behavior**

<kbd>J</kbd> to move down
<kbd>K</kbd> to move up

**Version**
demo at commit 0b555b9

**Screenshots**

Screen Recording 2022-06-13 at 11 21 18 AM mov

### PR DESCRIPTION
**Pull request by Luckey-Elijah**
## Status

**READY**

## Description

Fixes #413 

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [x] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by Luckey-Elijah** [comment]
Ready for review 🙂

File: packages/mason_logger/lib/src/mason_logger.dart [hunk 1]
```diff
@@ -223,11 +223,11 @@ class Logger {
       final byte = _stdin.readByteSync();
       if (event.length == 3) event.clear();
       event.add(byte);
-      if (_upKey.every(event.contains) || _jKey.every(event.contains)) {
```
Comment (Luckey-Elijah) [inline packages/mason_logger/lib/src/mason_logger.dart:226]: This pattern has appeared twice for checking how the index should be updated.

Perhaps we could change how this is written to reduce repeated code.

Comment (Luckey-Elijah) [reply packages/mason_logger/lib/src/mason_logger.dart:226]:
```dart
bool _eventIsOneOf(Iterable<int> event, Iterable<List<int>> keys) =>
  keys.any((key) => key.every(event.contains));
```
And use it like
```dart
if (_eventIsOneOf(event, [_upKey, _kKey])) {
  // handle up action
}
```

Comment (Luckey-Elijah) [reply packages/mason_logger/lib/src/mason_logger.dart:226]: Actually it's _6/7 times_ after taking a closer look

Comment (felangel) [reply packages/mason_logger/lib/src/mason_logger.dart:226]: yeah that sounds like an awesome improvement 👍

Comment (felangel) [reply packages/mason_logger/lib/src/mason_logger.dart:226]: You can even do it as an extension on Iterable<int> like:

```dart
extension on Iterable<int> {
  bool isOneOf(Iterable<int> items) {
    return items.any((item) => item.every(contains);
  }
}
```

And use like:

```dart
if (event.isOneOf([_upKey, _kKey])) {...}
```

Comment (Luckey-Elijah) [reply packages/mason_logger/lib/src/mason_logger.dart:226]: Yes, much easier to read 🙂

**Review by felangel — APPROVED** [review]

**Pull request merged**